### ISSUE TITLE
How to use this software?

### ISSUE THREAD

**Issue by TheStonemage** [0x192/Universal-Android-Debloater#174]
I dont understand how to set up the debloater app. Can i get someone to show me how to open it on my computer? I am really , really lost.

I saw the app advertised on youtube. I downloaded it but now that it is on my computer I don't understand any of this of how to get it to even start. There are no instructions. I am not a tech or computer guy but I thought I would ask for help. I am not tech saavy and this looks like its above me to know how to start it. I do know the usb debugging is open on my phone so i took care of that. But how do I get this going? Can we get instructions in laymans terms? I cant understand this.

**Comment by TheStonemage** [comment]
I saw the app advertised on youtube. I downloaded it but now that it is on my computer I don't understand any of this of how to get it to even start. There are no instructions. I am not a tech or computer guy but I thought I would ask for help. I am not tech saavy and this looks like its above me to know how to start it. I do know the usb debugging is open on my phone so i took care of that. But how do I get this going? Can we get instructions in laymans terms? I cant understand this.

**Comment by 0x192** [comment]
> Can we get instructions in laymans terms? I cant understand this.

Did you read the [`How to use`](https://github.com/0x192/universal-android-debloater#how-to-use-it=) section of the README? Can you tell me the steps you don't understand ? :slightly_smiling_face:

**Comment by TheStonemage** [comment]
The readme does not explain simply what to do if the program won't start. I did download it to my desktop but some things are not obvious. Such as.

1. Do we download this program to our phone as well?
2. When I open it all I see is a folder with files and have no idea what to to from there.  

i see . cargo, github, resources, src..a list of other stuff but nothing looks like what you guys have on this forum when i try to start it. That is is the biggest problem. How can i follow a readme if I dont understand the basic way of starting? Again your directions are seemingly set up for somebody who knows coding or is very computer saavy and sadly I am not it. 

That is the biggest problem what do I do to even open the program to start it? What do I click on first? Then from there what to do is just simply not explained easily so we know.  This normally happens when tech guys come up with programs like this, they are so advanced and smart that they at times forget we are all not on your level. I admit it, I am NOT there at all and I really want to use this that you have created in an effort to free myself from some apps I can't get off the phone any other way.  I hate the fact I can't remove amazon or google chrome. 
![Capturesddsds](https://user-images.githubusercontent.com/102837824/161429259-4a3f6d6e-747a-454c-bd96-cbf0a6ba1053.JPG)

**Comment by TheStonemage** [comment]
Above is what i see after the folder was downloaded to my desktop. My phone is already set to usb debugging so that part is ready.

**Comment by 0x192** [comment]
You downloaded the source code. That why you have a hard time!

The README redirects you to the release page:
> Download the latest release of UAD GUI for your Operating System [here](https://github.com/0x192/universal-android-debloater/releases). Take the opengl version only if the default version (with a Vulkan backend) doesn't launch.

Then, you just have to download  `uad_gui-windows.exe` or `uad_gui-windows-opengl.exe` in the **Assets** dropdown. To launch UAD, you only have to double click on the executable.

**Comment by 1337-537h** [comment]
I enabled developer options (im rooted)
I turn on USB Debugging
I have install device drivers and adb.
I have the program (0.5 uad_gui-windows.exe)
But the program wont connect to my phone.
Whats the problem?

**Comment by 0x192** [comment]
>  But the program wont connect to my phone.  Whats the problem?

I am not a magician. Don't you think I would need more information? Llike your logfile for example ([see the wik](https://github.com/0x192/universal-android-debloater/wiki/FAQ#where-are-the-logfiles=)).

Most of the time this issue happen because people didn't read the [`How to use`](https://github.com/0x192/universal-android-debloater#how-to-use-it=) section of the README and they forgot [to add adb to the PATH](https://www.architectryan.com/2018/03/17/add-to-the-path-on-windows-10/).

**Comment by 1337-537h** [comment]
I did everything.. the trick was to run it as admin. after that everything was ok.

**Comment by Cortru619** [comment]
The concept is great, Im pretty savvy with phones and computers and cant get it to connect. 
Is there a video on how to configure or set up? If I can figure out how to get it to connect and run I can def help by creating a visual solution and post it. 
I am trying to run it on my Note 20 ultra 5g and on a Note 8. 
My phones are ADB and debug ready via the computer and i also use wireless debugging for Shizuku.

**Comment by 1337-537h** [comment]
https://www.youtube.com/watch?v=-1LqYuvMKOA

Maybe this can help you.

**Comment by Cortru619** [comment]
Thanks! I will check it out!

**Comment by TheStonemage** [comment]
This too hard to do we need a step by step.Why can't it be loaded on the
phone instead of a PC?

**Comment by 0x192** [comment]
> This too hard to do we need a step by step

Again, can you tell me the steps you don't understand in the [How to use](https://github.com/0x192/universal-android-debloater#how-to-use-it=) section of the README? It's already a step by step guide.

> Why can't it be loaded on the phone instead of a PC?

Technical reason. You can't really run an adb server directly on the phone. There is [this nice project](https://github.com/tytydraco/LADB) but this is currently out of scope. I already have too much stuff to do with the current software.

**Comment by TheStonemage** [comment]
The readme still does not show me how to open the program. I can not get it
to even open!  Showmen the readme where it instructs you how.

**Comment by 0x192** [comment]
Huh? As I told you in a precedent message, you need to download the right executable in the Assets dropdown from the [release page](https://github.com/0x192/universal-android-debloater/releases). For you it's `uad_gui-windows.exe` or `uad_gui-windows-opengl.exe` (for older computer) .

To launch UAD, **you only have to double click on the executable**. I don't see how I can make it easier.

**Comment by TheStonemage** [comment]
I dont even see where to download.

Look i just dont have the brain power to understand this. Sorry I am not a
programmer.
This is way too hard to make work. It doesnt work./

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=icon>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=link>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by 1337-537h** [comment]
It works fine.
It is easy to make it work but the gui is not very user friendly.

This is an [alternative ](https://adbappcontrol.com/en/)easier to make it work with "better gui".
But it is not open source like universal-android-debloater.

I post above a [youtube link](https://www.youtube.com/watch?v=-1LqYuvMKOA) that may will help you run the program.

**Comment by cafuringa4** [comment]
After downloading it for windows, it didn't recognize my phone. If i move the exe to the actual adb platform-tools  folder, it works just fine without needing to add adb to the PATH. 

So, step by step for windows:

1. Enable Developer Options on your phone
2. on Developer Options, enable USB Debugging
3. connect your phone to the computer
4. enable usb debugging and file transfer on the popup on your phone
5. [download the Android SDK Platform Tools ZIP file for Windows](https://dl.google.com/android/repository/platform-tools-latest-windows.zip)
6. unzip the file
7. [download the universal android debloater for windows](https://github.com/0x192/universal-android-debloater/releases)
8. move the file to the platform-tools folder
9. run it from inside the folder

**Comment by Zupirio** [comment]
>8. move the file to the platform-tools folder
>9. run it from inside the folder

This part helped me run the program.

**Comment by cirramqc** [comment]
> > 8. move the file to the platform-tools folder
> > 9. run it from inside the folder
> 
> This part helped me run the program.

Same for me... That is what I needed to make it work :)  Thanks. for the great work

**Comment by TheStonemage** [comment]
I gave up. It won't work..doesn't work. Until this thing is loadable on a
phone, I can't see using it. Its just too hard for people to use. I'm not a
programmer.

**Comment by cirramqc** [comment]
> I gave up. It won't work..doesn't work. Until this thing is loadable on a phone, I can't see using it. Its just too hard for people to use. I'm not a programmer.

Just because you don't understand what you are doing, you shouldn't say that "it doesn't work"! If only 1 of 1000s says it doesn't work, then it might be an end user problem and not an app problem. 

Just because one sees something on YouTube doesn't always mean it's meant for everyone.  Possibly comment on the YouTube video or directly to it's creator to mention that he should possibly have specified that: "... you need to have some knowledge of inner workings of a mobile phone and installing unconventional programs, on a computer..."

As for "being loadable on a phone" that could be solved by rooting your phone. But if this is too complicated for you, I REALLY DON'T suggest you try rooting your phone. 

Maybe stick to apps you can find on Google Play.  Good luck to you.

**Comment by TheStonemage** [comment]
I gave up. This kind of stuff should be made where it's auto set up.
